controllers/datadogagent/feature/types.go

```diff
@@ -16,27 +17,69 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// ComponentsEnabled use to know which component need to be enabled for the feature
+// If set to:
+//   * true: the feature needs the corresponding component.
+//   * false: the corresponding component needs to ne disabled for this feature.
+//   * nil: the feature doesn't need the corresponding component.
+type ComponentsEnabled struct {
+	ClusterAgent       *bool
+	Agent              *bool
+	ClusterCheckRunner *bool
```

Comment:
i'm still wondering if we could/should have this be container-level rather than component-level. that is, keep the same structure but add a few lines for the agent containers. it would be the simplest form of the info we need..

Reply:
we discussed in IRL, feel free to ignore!